air/src/ir/mod.rs

```diff
@@ -22,6 +22,94 @@ pub use self::{
     value::{PeriodicColumnAccess, PublicInputAccess, Value},
 };
 
+/// A fixed two segment trace shape containing values for the main and aux segments.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct TraceShape<T> {
+    main: T,
+    aux: T,
```

Comment:
These could be `pub`, avoiding the need for the `get` and `get_mut` methods. The `Index` and `IndexMut` implementation would be used for indexing with a `TraceSegmentID`.